src/components/History/Components.js

```diff
@@ -29,49 +29,84 @@ ItemInfo.propTypes = {
 ItemInfo.defaultProps = {
   imgURL: reactJPG,
   title: 'The Great Gatsby',
-  status: 'Requesting',
-  supplement: 'Duration: 1-23 Mar. 2019'
+  status: 'Requesting'
+  // supplement: 'Duration: 1-23 Mar. 2019'
 };
 
 const RequestMessage = props => {
   const [id, setId] = useState(props.message.id);
+  const [showOperation, setShowOperation] = useState(
+    props.message.status === 'ongoing'
+  );
+  const [status, setStatus] = useState(props.message.status);
+  const {createdAt} = props.message.stuff
+    ? props.message.stuff
+    : {createdAt: '1970-01-01'};
+
   return (
     <div className="request-msg">
-      <div className="msg-time">{props.message.time}</div>
+      <div className="msg-time">
+        {createdAt
+          ? [new Date(createdAt)].map(
+              date =>
+                date.toDateString().substr(4) +
+                ', ' +
+                date.getHours() +
+                ':' +
+                date.getMinutes()
+            )[0]
+          : null}
+      </div>
       <div className="msg-body">
         <ItemInfo
-          imgURL={props.message.imgURL}
-          title={props.message.title}
-          status={props.message.status}
+          imgURL={
+            props.message.stuff
+              ? props.message.stuff.imageUrls.length > 0
```

Comment:
Let me tell you something cool. I think that his can be rewritten to something like the following:

``` props.message.stuff && props.message.stuff.imageUrls.length > 0 && props.message.stuff.imageUrls[0]```

I have to go right now, so I cannot test it, but maybe give it a try.

For reference see https://stackoverflow.com/questions/2932754/ternary-operators-in-javascript-without-an-else

Reply:
That's way clearer. Changed.